remove serializer='pickle' from celery tasks that are never passed arguments

nickpell opened the pull request:
Eventually we should stop using pickle as a celery serializer.  Doing so may help smooth out the python 3 migration, since django model objects pickled in python 2 have trouble being unpickled in python 3.

This is a quick pass at removing ```serializer='pickle'``` from functions that are never passed any arguments.

Commit:
remove serializer='pickle' from celery tasks that are never passed arguments

nickpell commented on corehq/apps/accounting/tasks.py:
```diff
@@ -239,7 +239,7 @@ def check_credit_line_balances():
 
-@periodic_task(serializer='pickle', run_every=crontab(hour=13, minute=0, day_of_month='1'), acks_late=True)
+@periodic_task(run_every=crontab(hour=13, minute=0, day_of_month='1'), acks_late=True)
 def generate_invoices(based_on_date=None):
```
argument is only used in testing

nickpell commented on corehq/apps/accounting/tasks.py:
```diff
@@ -623,13 +623,13 @@ def _fmt_row(sub):
 
-@periodic_task(serializer='pickle', run_every=crontab(minute=0, hour=0), queue='background_queue', acks_late=True)
+@periodic_task(run_every=crontab(minute=0, hour=0), queue='background_queue', acks_late=True)
 def update_exchange_rates(app_id=settings.OPEN_EXCHANGE_RATES_API_ID):
```
This function is only ever queued by celery beat - hence no arguments are passed.

nickpell commented on corehq/apps/accounting/tasks.py:
```diff
@@ -692,7 +692,7 @@ def assign_explicit_community_subscription(domain_name, start_date, method, acco
 
-@periodic_task(serializer='pickle', run_every=crontab(minute=0, hour=9), queue='background_queue', acks_late=True)
+@periodic_task(run_every=crontab(minute=0, hour=9), queue='background_queue', acks_late=True)
 def run_downgrade_process(today=None):
```
This function is only ever queued by celery beat - hence no arguments are passed.

nickpell commented on custom/icds_reports/tasks.py:
```diff
@@ -133,7 +133,7 @@
-@periodic_task(serializer='pickle', run_every=crontab(minute=30, hour=23),
+@periodic_task(run_every=crontab(minute=30, hour=23),
                acks_late=True, queue='icds_aggregation_queue')
 def run_move_ucr_data_into_aggregation_tables_task(date=None):
```
Only run as periodic task - never queue by our own code.

emord commented:
Overall I'm very pro making this type of change, but have a few questions.

What does this migration look like? If this is deployed as is, and there are tasks in the queue, will celery drop all tasks using the old serializer?

Also I'm against changing the the ones that do have arguments, even if they shouldn't have dates called in production, as it would be pretty easy to call those with a date later. Changes I would be comfortable are:

- Remove the parameter completely (for something that's used in tests, you could use `freezegun` if you need a set time)
- Using a custom json serializer that supports dates (not sure if that's possible, but I think Django provides one)
- changing them to accept a `date_string` and parsing that string to a date in the task

Commit:
don't pass an argument to update_exchange_rates

Commit:
don't pass an argument to run_downgrade_process

Commit:
leave pickle serializer on generate_invoices

Commit:
don't pass an argument to run_move_ucr_data_into_aggregation_tables_task

nickpell commented:
> What does this migration look like? If this is deployed as is, and there are tasks in the queue, will celery drop all tasks using the old serializer?

None of the tasks in the queue will get dropped when this is deployed.  They will get run per usual.

> Also I'm against changing the the ones that do have arguments

That's a good point.  I removed arguments from the tasks that never have them called, and reverted the change to ```generate_invoices``` until I decide on an approach for the 100+ remaining tasks.
